assets/scripts/dialogs/GeotagDialog.jsx

```diff
@@ -48,148 +44,138 @@ L.Icon.Default.mergeOptions({
-    if (!this.state.geocodeAvailable) return
-
+  const dispatch = useDispatch()
+  const initialState = getInitialState(props)
+  const [mapCenter, setMapCenter] = useState(initialState.mapCenter)
+  const [zoom, setZoom] = useState(initialState.zoom)
+  const [markerLocation, setMarkerLocation] = useState(
+    initialState.markerLocation
+  )
+  const [label, setLabel] = useState(initialState.label)
+  const [renderPopup, setRenderPopup] = useState(!!initialState.markerLocation)
+  const intl = useIntl()
+
+  const geocodeAvailable = !!PELIAS_API_KEY
+
+  // `dpi` is a bad name for what is supposed to be referring to the devicePixelRatio
+  // value. A devicePixelRatio higher than 1 (e.g. Retina or 4k monitors) will load
+  // higher resolution map tiles.
+  const dpi = useSelector((state) => state.system.devicePixelRatio || 1.0)
+  const tileUrl = dpi > 1 ? MAP_TILES_2X : MAP_TILES
+
+  const handleMapClick = (event) => {
+    // get the new latlng of the clicked position
     const latlng = {
       lat: event.latlng.lat,
       lng: event.latlng.lng
     }
 
-    // this is in the context of a a map click, we don't want to switch up
+    // this is in the context of a a map click or drag, we don't want to switch up
     // the zoom level on the user
-    // (or at least im assuming this based on the code ;-))
     const zoom = event.target.getZoom()
-
-    this.reverseGeocode(latlng).then((res) => {
+    reverseGeocode(latlng).then((res) => {
       const latlng = {
         lat: res.features[0].geometry.coordinates[1],
         lng: res.features[0].geometry.coordinates[0]
       }
-
-      this.setState({
-        mapCenter: latlng,
-        zoom: zoom,
-        renderPopup: true,
-        markerLocation: latlng,
-        label: res.features[0].properties.label,
-        bbox: res.bbox || null
-      })
-
-      this.props.setMapState({
-        markerLocation: latlng,
-        addressInformation: res.features[0].properties
-      })
+      setZoom(zoom)
+      updateMap(latlng, res.features[0].properties, res.features[0].label)
     })
   }
 
-  handleMarkerDragEnd = (event) => {
-    const latlng = event.target.getLatLng()
-
-    this.reverseGeocode(latlng).then((res) => {
-      this.setState({
-        renderPopup: true,
-        markerLocation: latlng,
-        label: res.features[0].properties.label,
-        bbox: res.bbox || null
-      })
-
-      this.props.setMapState({
-        markerLocation: latlng,
-        addressInformation: res.features[0].properties
-      })
-    })
+  const handleMarkerDragStart = (event) => {
+    setRenderPopup(false)
   }
 
-  handleMarkerDragStart = (event) => {
-    this.setState({
-      renderPopup: false
+  const handleMarkerDragEnd = (event) => {
+    /* method of updating coords is different here(as opposed to onClick) since we're
+     moving a leaflet object
+    keeping the data shape of latlng in line with how its used in child components
+    */
```

Comment:
For context, we settled on this because whether it's `[lat, lng]` or `[lng, lat]` is really annoying for anyone who isn't steeped in geo and/or Leaflet. We made a decision to be explicit by passing around an object with labelled properties. It's definitely not common in Leaflet world because of how verbose it can be, but it's supported, and we picked it.

Reply:
That makes sense! I also ran into a PropType validation error when I started 'cleaning' it up, so I figured this note might help prevent someone from doing the same thing at a later date. :-D